File: 49.group-anagrams/solution.cpp
class Solution {
public:
	vector<vector<string>> groupAnagrams(vector<string>& strs) {
		map<string, vector<string>> cache;
		for (vector<string>::const_reference str : strs)
		{
			string s = str;
			sort(s.begin(), s.end());
			cache[s].push_back(str);
		}
		vector<vector<string>> result;
		for (map<string, vector<string>>::const_reference r : cache)
			result.push_back(r.second);
		return result;
	}
};

File: 49.group-anagrams/solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dict = {}
        for s in strs:
            k = ''.join(sorted(list(s)))
            if k not in dict:
                dict[k] = [s]
            else:
                dict[k].append(s)
        return list(dict.values())
